stable/external-dns/templates/deployment.yaml

```diff
@@ -327,13 +327,16 @@ spec:
       # Azure volume(s)
       {{- if eq .Values.provider "azure" }}
       - name: azure-config-file
-        {{- if (not .Values.azure.secretName) }}
+        {{- if .Values.azure.secretName }}
+        secret:
+          secretName: {{ .Values.azure.secretName }}
+        {{- else if (and .Values.azure.resourceGroup .Values.azure.tenantId .Values.azure.subscriptionId .Values.azure.aadClientId .Values.azure.aadClientSecret (not .Values.azure.useManagedIdentityExtension)) (and .Values.azure.resourceGroup .Values.azure.tenantId .Values.azure.subscriptionId .Values.azure.useManagedIdentityExtension) }}
```

Comment:
Wouldn't be easier to just check the mandatory parameters?
```diff
- {{- else if (and .Values.azure.resourceGroup .Values.azure.tenantId .Values.azure.subscriptionId .Values.azure.aadClientId .Values.azure.aadClientSecret (not .Values.azure.useManagedIdentityExtension)) (and .Values.azure.resourceGroup .Values.azure.tenantId .Values.azure.subscriptionId .Values.azure.useManagedIdentityExtension) }}
+ {{- else if (and .Values.azure.resourceGroup .Values.azure.tenantId .Values.azure.subscriptionId }}
```
Then you can add a `{{- define "external-dns.validateValues.azureConfiguration }}` function in the helpers (check how we did it for other "external-dns.validateValues" functions) to ensure the `addClientId` and `addClientSecret` are not set when `useManagedIdentityExtension` is set, and viceversa.

Reply:
All right, took me a while to figure out how to do that, but I just pushed the changes :+1: Cool that you can do validation in helm such way, TIL.